lib/Cake/Model/Datasource/DboSource.php

```diff
@@ -3459,12 +3469,16 @@ public function buildColumn($column) {
 			return null;
 		}
 
-		if (!isset($this->columns[$type])) {
-			trigger_error(__d('cake_dev', 'Column type %s does not exist', $type), E_USER_WARNING);
-			return null;
+		// if a storage requirement in bytes was set, try it first (i.e. 'integer/4' before 'integer')
+		if (isset($storage) && isset($this->columns[$type . '/' . $storage])) {
+			$real = $this->columns[$type . '/' . $storage];
```

Comment:
Because there is unfortunately no API in `DboSource::buildColumn($column)` that given *all* the fields in `$column` that describe the column in an abstract way (i.e. `type` *and* the new `storage` field) would return the proper real/native type like `smallint`. There *is* such method when going the other way around -- `DataSource::column()` takes a real column type and return an abstract type.

Instead, `buildColumn()` is *directly* poking at `$this->columns[$type]`, an *array* that apparently all datasources subclasses are supposed to have in place. What this means to me is that I couldn't introduce a new API like `DboSource::abstractToRealColumn()` (the opposite of `column()`) which would take the whole `$column` array of fields as a parameter instead of just `$type`, because if people have implemented their own data sources with a `$this->columns[]` array as they are supposed to, then these will no longer work, right? 

I can't introduce a function call there if there was none in the first place, there is no contract to be enforced -- datasource subclasses were never meant to call a `abstractToRealColumn()` function. So with that in mind I asked myself "how can I still use the `$this->columns[]` array and allow developers to specify a syntax there that maps an abstract type *and* a storage requirement *to* a native type?". That's why I went for a string representation that would just be the type followed by '/' followed by the storage. Does it make sense?

Alternatively I could do both. Introduce a new `abstractToRealColumn()` function that takes `$column`, not just `$type`, and if that method is not found then use the old `$this->columns` array?